packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -55,8 +55,9 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
 
     this.convertPriceFeedDecimals = number => {
       // Converts price result to wei
-      // returns price conversion to correct priceFeedDecimals as a big number
-      return this.toBN(parseFixed(number.toString(), priceFeedDecimals).toString());
+      // returns price conversion to correct decimals as a big number.
+      // Note: Must ensure that `number` has no more decimal places than `priceFeedDecimals`.
+      return this.toBN(parseFixed(number.toString().substring(0, priceFeedDecimals), priceFeedDecimals).toString());
```

Comment:
why are you doing this substring?

Reply:
Because sometimes the price returned by the CW API has a lot of decimals, and if the # of decimals > `priceFeedDecimasl` this throws an error. 

For example say I want 6 decimal precision, but the CW API returns a price with 7 decimals. This will be an error. This is really only an issue with REST API price feeds, not Balancer/Uniswap for example which return smart contract data

Reply:
Ive found using toFixed(this.decimals) usually does the trick